tests/stream_test.ts

```diff
@@ -0,0 +1,810 @@
+    assertEquals(fullStreamInfo.length, 2);
+    assert(fullStreamInfo.radixTreeKeys > 0);
+    assert(fullStreamInfo.radixTreeNodes > 0);
+    assertEquals(fullStreamInfo.groups.length, 1);
+    assertEquals(fullStreamInfo.groups[0].consumers.length, 1);
+
+    const cc = fullStreamInfo.groups[0].consumers[0];
+    assertEquals(cc.name, "someone");
+    assert(cc.seenTime > 0);
+    assertEquals(cc.pelCount, 2);
+    assertEquals(cc.pending.length, 2);
+    for (const msg of cc.pending) {
+      assertEquals(msg.timesDelivered, 1);
+    }
+    assertEquals(fullStreamInfo.entries.length, 2);
+
+    const limitWithCount = await client.xinfo_stream_full(key, 1);
+    assertEquals(limitWithCount.length, 2);
+    assert(limitWithCount.radixTreeKeys > 0);
+    assert(limitWithCount.radixTreeNodes > 0);
+    assertEquals(limitWithCount.groups.length, 1);
+    assertEquals(limitWithCount.groups[0].consumers.length, 1);
+
+    const c = limitWithCount.groups[0].consumers[0];
+    assertEquals(c.name, "someone");
+    assert(c.seenTime > 0);
+    assertEquals(c.pelCount, 2);
+    // The COUNT option limits this array to a single entry!
+    assertEquals(c.pending.length, 1);
+    for (const msg of c.pending) {
+      assertEquals(msg.timesDelivered, 1);
+    }
+    // The COUNT option limits this array to a single entry!
+    assertEquals(limitWithCount.entries.length, 1);
+
+    // Let's make another group and see more stats
+    await client.xgroup_create(key, "newgroup", "$", true);
+
+    const groupInfos = await client.xinfo_groups(key);
+    assertEquals(groupInfos.length, 2);
+
+    console.log(JSON.stringify(groupInfos));
+    const newGroup = groupInfos.find((g) => g.name === "newgroup");
+    const oldGroup = groupInfos.find((g) => g.name === group);
+    assert(newGroup);
+    assert(oldGroup);
+
+    assertEquals(oldGroup.pending, 2);
+    assertEquals(newGroup.pending, 0);
+
+    // Add one more record and read it with a new consumer,
+    // so that we can check the parsing of deno-redis xinfo_consumers
+    await client.xadd(key, "*", { "hello": "maybe" });
+    await client.xreadgroup([[key, ">"]], { group, consumer: "newbie" });
+
+    // Increase the idle time by falling asleep
+    await sleepMillis(2);
+    const consumerInfos = await client.xinfo_consumers(key, group);
+    assertEquals(consumerInfos.length, 2);
+    const newConsumer = consumerInfos.find((c) => c.name === "newbie");
+    const oldConsumer = consumerInfos.find((c) => c.name === "someone");
+    assert(newConsumer);
+    assert(oldConsumer);
+    assert(newConsumer.idle > 1);
+    assert(oldConsumer.idle > 1);
+    // New consumer read one message with ">"
+    assertEquals(newConsumer.pending, 1);
+    // Old consumer read two messages with ">"
+    assertEquals(oldConsumer.pending, 2);
+
+    assertEquals(await client.xgroup_destroy(key, "newgroup"), 1);
+  });
+});
+
+const sleepMillis = (ms: number) => {
```

Comment:
Please use `defer` (defined in `std/async/mod.ts`) instead.

```suggestion
import { defer as sleepMillis } from "../vendor/https/deno.land/std/async/mod.ts";
```